src/Expressions/Literals/Map.php

```diff
@@ -38,29 +35,26 @@ final class Map implements MapType
     private array $elements;
 
     /**
-     * @param AnyType[] $elements Associative array of the elements that this map should have
+     * @param mixed[] $elements Associative array of the elements that this map should have
+     *
      * @internal This method is not covered by the backwards compatibility promise of php-cypher-dsl
      */
     public function __construct(array $elements = [])
     {
-        self::assertClassArray('elements', AnyType::class, $elements);
-        $this->elements = $elements;
+        $this->elements = array_map([self::class, 'toAnyType'], $elements);
```

Comment:
This works already, because one of the traits in the hierarchy (`AnyTypeTrait`) uses the `CastTrait`. But I agree it is better to include it directly as well.